lib/project/util.coffee

```diff
@@ -36,7 +36,21 @@ showIf = (condition) ->
   else
     {display: 'none'}
 
+capitalize = (str) -> str.replace(/(?:^|\s)\S/g, (capital) -> capital.toUpperCase())
+
+preserveCase = (text, reference) ->
+  # If replaced text is capitalized (strict), capitalize replacement
+  if reference is capitalize(reference.toLowerCase())
+    capitalize(text)
+
+  # If replaced text is uppercase, uppercase replacement
+  else if reference is reference.toUpperCase()
+    text.toUpperCase()
+  else
+    text
```

Comment:
This seems like a difficult feature to design. You have implemented preservation of All CAPS, and of Proper case. I would have expected lower case to be preserved as well. It might be worth thinking through a few more cases, such as strings with chars other than spaces and letters (e.g. one could imaging replacing snake_case and SNAKE_CASE at the same time, one being recognizably lower case and the other upper). camelCase might be interesting too if both the find and replace had that form, although maybe now I'm getting too fancy. I would prefer to open this discussion a little bit, though, such as with an issue for this feature.

Reply:
You have support for 

- Title Case
- UPPER CASE

I suggest you add

- lower case
- Sentence case

These are the main ones in [Wikipedia/Letter_case](https://en.wikipedia.org/wiki/Letter_case), and ones that can be recovered from lower case. Then the user can set the default (what they put in the replace box) to whatever mixed case they want otherwise. I realize not supporting "Sentence case" was by design, but I would expect it to keep that case. Thanks!